beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -619,7 +620,8 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                     return Err(CouplingError::DataColumnPeerFailure {
                         error: format!("Peers did not return column for block_root {block_root:?} {naughty_peers:?}"),
                         faulty_peers: naughty_peers,
-                        action: PeerAction::LowToleranceError,
+                        // The block peer might be malcicious so don't downscore the column peer too bad
+                        action: PeerAction::MidToleranceError,
```

Comment:
For mainnet this should be no penalty at all. Otherwise, it can still be exploited. If peers are naughty we should not request from them, but not downscore

Reply:
I feel like there should be some penalty to actually get us to notice this. From my observation, majority of the times we get a Coupling or a LookupVerifyError, its because of a client bug rather than something malicious or the node not having something on a different chain. I think the downscoring is an extremely valuable signal.

I think a malicious block peer completely eclipsing us reliably and at a scale that matters is extremely hard to do because of the way our peer pruning logic works. I would prefer we have some guardrails against the majority case of a buggy client rather than no penalty. Made 9b2de09 to make everything a HighTolerance error. Lmk what you think